test/hashing.jl

```diff
@@ -310,3 +310,6 @@ struct AUnionParam{T<:Union{Nothing,Float32,Float64}} end
 @test Type{AUnionParam{<:Union{Nothing,Float32,Float64}}} === Type{AUnionParam}
 @test Type{AUnionParam.body}.hash == 0
 @test Type{Base.Broadcast.Broadcasted}.hash != 0
+
+# Issue #50065
+@test hash(1.0) == hash(BigFloat(1.0))
```

Comment:
add `1//1` and `big(1)` for good measure.

Reply:
I think it makes sense to add BigFloat and BigInt to the massive cross-product at the top instead.

Reply:
well I'm sure glad I suggested testing `BigInt` :)